### PR TITLE
Support assigning a vector-valued Constant to a mixed Function

### PR DESCRIPTION
**Pull request by wence-**
If the Constant has the same shape as the mixed function's element,
split it into as many pieces as there are spaces in the mixed space and
use each of these in turn to assign to the sub functions.

### CODE REVIEW

**Comment by wence-** [comment]
Addresses #471 afaict.

File: firedrake/assemble_expressions.py [hunk 1]
```diff
@@ -368,6 +368,28 @@ def terminal(self, o):
...
+                consts = []
+                val = o.dat.data_ro
+                for fs in self._fs:
+                    shp = fs.ufl_element().value_shape()
+                    if len(shp) == 0:
+                        c = constant.Constant(val[offset], domain=o.domain())
+                        offset += 1
+                    elif len(shp) == 1:
+                        c = constant.Constant(val[offset:offset+shp[0]],
+                                              domain=o.domain())
```
Comment (kynan) [inline firedrake/assemble_expressions.py]: Wouldn't you also need to increase the offset in this case?

Comment (wence-) [inline firedrake/assemble_expressions.py]: Oh yes.

**Comment by kynan** [comment]
I think this is good!

**Pull request merged**